NETWORK_PROTOCOL.md

```diff
@@ -1 +1,102 @@
-Placeholder for network protocol.
+Status: this is a rough draft, intended to get some ideas out there. 
+
+Missing: how to run on top of HTTP/2. Possibly the structure or semantics of publisher name strings.
+Might be removed: extension support.
+Need to choose serialization method (protobuf, msgpack, other?)
```

Comment:
I'm suggesting we pick one and mandate it. This is for serializing the framing and the protocol messages themselves, not the inner payloads. The payloads are necessarily opaque to the protocol, but the user can choose to use the same serialization format for the payloads for synergy.